src/box/lua/console.c

```diff
@@ -886,7 +886,7 @@ tarantool_lua_console_init(struct lua_State *L)
 		{"run_on_eval",		lbox_console_run_on_eval},
 		{NULL, NULL}
 	};
-	luaT_newmodule(L, "console", consolelib);
+	luaT_newmodule(L, "console.lib", consolelib);
```

Comment:
Quoted @igormunkin's comments to discuss in it a thread:

> I am confused with the lib name. There is a patch were you start using `internal.print` for quite the similar reason, and now you decided to use `lib` suffix. Could you please share the stats regarding `lib` usage you've mentioned in the commit message?

> **[PATCH 5/6] lua: don't tweak package.loaded.compat**
> <...>
>
> Anyway, still confused with the naming. Why is `internal` prefix used here?

Reply:
To make it clear, my current approach is the following.

1. Use `internal.foo` for a Lua module that is not exposed into the public API. Existing examples: `internal.argparse`, `internal.trigger`, `internal.print` (the latter is mine).
2. Use `foo.lib` for a Lua/C module that is not exposed into the public API and used in the `foo` module. Existing examples: `net.box.lib`. Hm, just one.

The compat module is a good example. It is not exposed directly as a module (`require('compat')` shouldn't return it). And it has both Lua and Lua/C parts. So the Lua part is named `internal.compat` according to the rule 1. And so the Lua/C part is named `internal.compat.lib` according to the rule 2.

The hack with the same named Lua/C and Lua modules was the most used approach. Now I need something else and I stick with what I saw around. It is self-consistent at least.

If there is another proposal, I'm open for it.

Reply:
OK, that works for me. Anyway, it would be great to see this in our style guide (or something similar).

Reply:
I filed a stub issue: https://github.com/tarantool/doc/issues/3308.